raw_decoder/raw_decoder.go

```diff
@@ -0,0 +1,317 @@
+package raw_decoder
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"golbat/grpc"
+	"golbat/pogo"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type RawDecoder interface {
+	GetProtoDataFromHTTP(http.Header, []byte) (*ProtoData, error)
+	GetProtoDataFromGRPC(*grpc.RawProtoRequest) *ProtoData
+	Decode(context.Context, *ProtoData, func(context.Context, *Proto))
+}
+
+type ProtoData struct {
+	*CommonData
+	Protos []Proto
+}
+
+func (pd ProtoData) Lat() float64 {
+	l := len(pd.Protos)
```

Comment:
I've not followed exactly when this is called - for the standard MITM send, the lat/lon at the top of the body indicates where the calls were made from (and where processing rules are applied against); individual lat/lon on items aren't really used. I know this is what pogodroid does, but this is the exception I think.

Reply:
I also believe PD to be the exception, but since this object is the general object being built by the /raw decoder, it sort of needs to support it. Although, I don't remember how this is being used either. I will look it over.